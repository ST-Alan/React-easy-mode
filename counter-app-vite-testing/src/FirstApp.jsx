import PropTypes from 'prop-types'
// import { Fragment } from "react"

// export const FirstApp = () => {
// 	return (
// 		<Fragment>
// 			<h2>First App</h2>
// 			<p>Second Line</p>
// 		</Fragment>
// 	)
// }
// El Fragment de arriba es igual al <></> de abajo y sin importar

export const FirstApp = ({ title, subTitle, name }) => {
	return (
		<>
			<h1>{title}</h1>
			<p>IOS & Android version {subTitle}</p>
			<p>{name}</p>
		</>
	)
}

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.number
}

FirstApp.defaultProps = {
	// title: 'App Buyxpress Espa√±ol',
	subTitle: 33,
	name: 'Alan Fermin'
}