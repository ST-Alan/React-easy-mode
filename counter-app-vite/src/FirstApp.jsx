import PropTypes from 'prop-types'
// import { Fragment } from "react"

// export const FirstApp = () => {
// 	return (
// 		<Fragment>
// 			<h2>First App</h2>
// 			<p>Second Line</p>
// 		</Fragment>
// 	)
// }
// El Fragment de arriba es igual al <></> de abajo y sin importar

export const FirstApp = ({title,subTitle}) => {
	return (
		<>
			<h2>First App {title}</h2>
			<p>IOS & Android version {subTitle}</p>
		</>
	)
}

FirstApp.propTypes={
	title: PropTypes.string,
	subTitle: PropTypes.number
}

FirstApp.defaultProps={
	title: 'App Buyxpress Espa√±ol',
	subTitle: 'Version 0.0.1'
}