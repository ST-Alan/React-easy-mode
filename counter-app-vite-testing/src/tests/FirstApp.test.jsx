import { render } from "@testing-library/react";
import { FirstApp } from "../FirstApp";
describe('Pruebas en FirstApp', () => {

	// test('Debe de hacer match con el snapshot', () => {

	// 	const title = 'Hola, Soy Goku'
	// 	const { container } = render(<FirstApp title={title} />)
	// 	console.log(container)
	// 	expect(container).toMatchSnapshot()


	// })

	test('Debe mostrar el título en un h1', () => {
		const title = 'Hola, Soy Goku'
		// La desestructuración con jest permite colocar otro método después del container. en este caso se busca obtener un nodo por un texto, entonces se usa getByText
		const { container, getByText } = render(<FirstApp title={title} />)

		// El getByText lo que hace es que dentro de lo que se renderiza se encuentre el título
		// El toBeTruthly() quiere indicar que ese elemento debe existir
		expect(getByText(title)).toBeTruthy()

		const h1 = container.querySelector('h1')
		//Lo que estoy esperando es que el h1.innerTML sea la constante title dentro de una etiqueta h1
		// expect(h1.innerHTML).toBe(title)
		// Se puede hacer como la línea de arriba o se puede hacer como la línea de abajo
		expect(h1.innerHTML).toContain(title)

	})

})