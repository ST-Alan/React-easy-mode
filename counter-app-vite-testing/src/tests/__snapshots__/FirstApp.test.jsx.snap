// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    IOS & Android version 
    33
  </p>
  <p>
    Alan Fermin
  </p>
</div>
`;
